crates/bevy_picking/src/mesh_picking/mod.rs

```diff
@@ -0,0 +1,132 @@
+//! marked with the [`RayCastPickable`] component.
+//!
+//! To manually perform mesh ray casts independent of picking, use the [`MeshRayCast`] system parameter.
+
+pub mod ray_cast;
+
+use crate::{
+    backend::{ray::RayMap, HitData, PointerHits},
+    prelude::*,
+    PickSet,
+};
+use bevy_app::prelude::*;
+use bevy_ecs::prelude::*;
+use bevy_reflect::prelude::*;
+use bevy_render::{prelude::*, view::RenderLayers};
+use ray_cast::{MeshRayCast, RayCastSettings, RayCastVisibility, SimplifiedMesh};
+
+/// Runtime settings for the [`MeshPickingBackend`].
+#[derive(Resource, Reflect)]
+#[reflect(Resource, Default)]
+pub struct MeshPickingBackendSettings {
+    /// When set to `true` ray casting will only happen between cameras and entities marked with
+    /// [`RayCastPickable`]. Off by default.
+    ///
+    /// This setting is provided to give you fine-grained control over which cameras and entities
+    /// should be used by the ray cast backend at runtime.
+    pub require_markers: bool,
+
+    /// When set to [`RayCastVisibility::Any`], hidden meshes can be ray casted against.
+    ///
+    /// See [`RayCastSettings::visibility`] for more information.
+    pub ray_cast_visibility: RayCastVisibility,
+}
+
+impl Default for MeshPickingBackendSettings {
+    fn default() -> Self {
+        Self {
+            require_markers: false,
+            ray_cast_visibility: RayCastVisibility::VisibleInView,
+        }
+    }
+}
+
+/// An optional component that marks cameras and target entities that should be used in the ray cast picking backend.
+/// Only needed if [`MeshPickingBackendSettings::require_markers`] is set to `true`.
+#[derive(Debug, Clone, Default, Component, Reflect)]
+#[reflect(Component, Default)]
+pub struct RayCastPickable;
+
+/// Adds the mesh picking backend to your app.
+#[derive(Clone, Default)]
+pub struct MeshPickingBackend;
+
+impl Plugin for MeshPickingBackend {
+    fn build(&self, app: &mut App) {
+        app.init_resource::<MeshPickingBackendSettings>()
+            .register_type::<(RayCastPickable, MeshPickingBackendSettings, SimplifiedMesh)>()
+            .add_systems(PreUpdate, update_hits.in_set(PickSet::Backend));
+    }
+}
+
+/// Casts rays into the scene using [`MeshPickingBackendSettings`] and sends [`PointerHits`] events.
+#[allow(clippy::too_many_arguments)]
+pub fn update_hits(
+    backend_settings: Res<MeshPickingBackendSettings>,
+    ray_map: Res<RayMap>,
+    picking_cameras: Query<(&Camera, Option<&RayCastPickable>, Option<&RenderLayers>)>,
+    pickables: Query<&PickingBehavior>,
+    marked_targets: Query<&RayCastPickable>,
+    layers: Query<&RenderLayers>,
+    mut ray_cast: MeshRayCast,
+    mut output: EventWriter<PointerHits>,
+) {
+    for (&ray_id, &ray) in ray_map.map().iter() {
+        let Ok((camera, cam_pickable, cam_layers)) = picking_cameras.get(ray_id.camera) else {
+            continue;
```

Comment:
I probably wouldn't want it to spam errors every frame, so it'd need to be tracked in some way so that it only logs it once. Either way, I don't think that error/warning belongs here; it should be handled higher up in the picking logic where the `RayMap` is managed. Otherwise you could get a situation where every backend emits its own duplicate warnings.

I also don't think it's even really possible for rays to exist in the `RayMap` without a camera, since the `RayMap` is cleared and repopulated every frame based on pointer positions, which requires cameras.